### DISCUSSION TITLE
Is it possible to pull new commits?

### DISCUSSION THREAD

**Discussion by cooperspencer** [GitoxideLabs/gitoxide#628] [Q&A] [answered]
Hi,
sorry if it is a stupid question, but as it is possible to clone repositories now, is it also somehow possible to pull the latest changes?

**Comment by Byron** [comment, marked as answer]
No worries, this is a "only stupid answers" zone ;).

A pull is `git fetch && git merge`, and since there is no `gix merge` yet one would probably do `gix fetch && git merge` to get something similar to `git pull`.

**Comment by 38github** [comment]
I believe it would be a good idea to write about lacking features on the first page and include this. I almost went mad trying to find out how to pull but fortunately I found this Q&A.

**Reply by Byron** [reply]
Please note that it's still early days for `gitoxide` and many features are missing to reach feature parity even with `git2`. The resource to know what's implemented is either…

* [`crate-status.md`](https://github.com/Byron/gitoxide/blob/main/crate-status.md)
* [`gix` docs on docs-rs with the `git2` search term](https://docs.rs/gix/0.54.1/gix/?search=git2)

With that, it's possible to probably discover what you are looking for or not, which would be a good indication of it not being present, or planned if it shows up in the `crate-status.md`.

I also have [added a new paragraph](https://github.com/Byron/gitoxide#feature-discovery) with the above and a high-level overview of typical features.
